Lib/test/libregrtest/runtest.py

```diff
@@ -284,34 +337,40 @@ def _runtest_inner(ns: Namespace, test_name: str, display_failure: bool = True)
     except support.ResourceDenied as msg:
         if not ns.quiet and not ns.pgo:
             print(f"{test_name} skipped -- {msg}", flush=True)
-        return RESOURCE_DENIED
+        return ResourceDenied(test_name)
     except unittest.SkipTest as msg:
         if not ns.quiet and not ns.pgo:
             print(f"{test_name} skipped -- {msg}", flush=True)
-        return SKIPPED
+        return Skipped(test_name)
+    except support.TestFailedWithDetails as exc:
+        msg = f"test {test_name} failed"
+        if display_failure:
+            msg = f"{msg} -- {exc}"
+        print(msg, file=sys.stderr, flush=True)
+        return Failed(test_name, errors=exc.errors, failures=exc.failures)
     except support.TestFailed as exc:
```

Comment:
In what scenarios are we are raising `Failed` but not `TestFailedWithDetails` now?

Reply:
There are quite a few bare TestFailed in custom `test_main` functions in various `test_*.py` files. Example:

https://github.com/python/cpython/blob/c05a790693b3b00ac7cb5b96ad416ca9d8ecb14a/Lib/test/test_ssl.py#L5044